lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -302,25 +291,79 @@ void SessionStateEnforcer::m5g_complete_termination(
   return;
 }
 
+void SessionStateEnforcer::m5g_state_change_action(
+    std::string& imsi, SessionStateUpdateCriteria& session_uc,
+    SetSmNotificationContext notif, std::unique_ptr<SessionState>& session) {
+  /*Irrespective of any State of Session, release and terminate*/
+  if ((notif.common_context().sm_session_state() == ACTIVE_2) &&
+      (session->get_state() == INACTIVE)) {
+    MLOG(MINFO) << "Active state transition for session id: "
+                << session->get_session_id();
+    session->set_fsm_state(ACTIVE, session_uc);
+    uint32_t cur_version = session->get_current_version();
+    session->set_current_version(++cur_version, session_uc);
+    MLOG(MINFO) << "During Active state of session changed to "
+                << session_fsm_state_to_str(session->get_state());
+    /* Call for all rules to be associated from session
+     * and inform to UPF
+     */
+    m5g_add_rules_and_update_upf(imsi, session);
+  }
+  /* Now code for Inactive state notificaiton
+   */
+  else if (
+      (notif.common_context().sm_session_state() == INACTIVE_3) &&
+      (session->get_state() == ACTIVE)) {
+    MLOG(MINFO) << "Inactive state transition for session id"
+                << session->get_session_id();
+    session->set_fsm_state(INACTIVE, session_uc);
+    uint32_t cur_version = session->get_current_version();
+    session->set_current_version(++cur_version, session_uc);
+    MLOG(MINFO) << "During InActive state of session changed to "
+                << session_fsm_state_to_str(session->get_state());
+    /* Call for all rules to be de-associated from session
+     * and inform to UPF
+     */
+    m5g_remove_rules_and_update_upf(imsi, session);
+  }
+}
+

```

Comment:
```
void SessionStateEnforcer::m5g_state_change_action(
    std::string& imsi, SessionStateUpdateCriteria& session_uc,
    SetSmNotificationContext notif, std::unique_ptr<SessionState>& session) {
  if ((notif.common_context().sm_session_state() == ACTIVE_2) &&
      (session->get_state() == INACTIVE)) {
	m5g_execute_state_change_action(imsi, session_uc, notif, session, ACTIVE)
  }
  else if (
      (notif.common_context().sm_session_state() == INACTIVE_3) &&
      (session->get_state() == ACTIVE)) {
	m5g_execute_state_change_action(imsi, session_uc, notif, session, INACTIVE)
  }
}

void SessionStateEnforcer::m5g_execute_state_change_action(
    std::string& imsi, SessionStateUpdateCriteria& session_uc,
    SetSmNotificationContext notif, std::unique_ptr<SessionState>& session, 
 	SessionFsmState targetState) {
	auto stateStr = session_fsm_state_to_str(targetState)
    MLOG(MINFO) << stateStr <<" state transition for session id"
                << session->get_session_id();
    session->set_fsm_state(targetState, session_uc);
    uint32_t cur_version = session->get_current_version();
    session->set_current_version(++cur_version, session_uc);
    MLOG(MINFO) << "During "<< stateStr <<" state of session changed to "
                << session_fsm_state_to_str(session->get_state());
    /* Call for all rules to be de-associated from session
     * and inform to UPF
     */
    m5g_remove_rules_and_update_upf(imsi, session);

}

```

you need to update `std::string session_fsm_state_to_str(SessionFsmState state)` to include INACTIVE

Reply:
Aren't you using a fork of magma?

Reply:
Well , I fork my own repo which is created 2 months, however my upstream master is latest magma repo. That is the procedure I am following.

Reply:
I did slight change and adapted, only difference is PDR rules to be installed/removed are changed.

Reply:
I have adapted to most of the extent.